internal/storage/v2/clickhouse/client/client.go

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) 2025 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package client
+
+import (
+	"context"
+
+	"go.opentelemetry.io/collector/pdata/ptrace"
+)
+
+type ChPool interface {
+	Do(ctx context.Context, query string, td ptrace.Traces) error
```

Comment:
DB API abstraction should not be dependent on a business-specific data model. It should accept Input or something DB-specific.